hypershift/hosted-wrapper.py

```diff
@@ -0,0 +1,592 @@
+    ocm_login_command = [ocm_cmnd, "login", "--url=" + args.ocm_url, "--token=" + args.ocm_token]
+    logging.debug(ocm_login_command)
+    ocm_login_process = subprocess.Popen(ocm_login_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    ocm_login_stdout,ocm_login_stderr = ocm_login_process.communicate()
+    if ocm_login_process.returncode != 0:
+        logging.error('%s unable to execute `ocm login`' % ocm_cmnd)
+        logging.error(ocm_login_stderr.strip().decode("utf-8"))
+        exit(1)
+    else:
+        logging.info('`ocm login` execution OK')
+        logging.debug(ocm_login_stdout.strip().decode("utf-8"))
+
+    try:
+        with open(args.pull_secret_file, "r") as pull_secret_file:
+            json.load(pull_secret_file)
+            logging.info('%s is a valid json file' % args.pull_secret_file)
+    except Exception as err:
+        logging.error(err)
+        logging.error('Failed to read pull secret file %s' % args.pull_secret_file)
+        exit(1)
+
+    mgmt_metadata = _get_mgmt_cluster_info(ocm_cmnd, args.mgmt_cluster,es,args.es_index,args.es_index_retry,my_uuid,args.cluster_count,args.workers)
+
+    if 'cluster_id' not in mgmt_metadata or 'base_domain' not in mgmt_metadata or 'aws_region' not in mgmt_metadata:
+        logging.error('Failed to obtain Management Cluster information from %s' % args.mgmt_cluster)
+        exit(1)
+    else:
+        logging.debug('Management Cluster information for %s:' % mgmt_metadata['cluster_name'])
+        logging.debug('             Custer ID:   %s' % mgmt_metadata['cluster_id'])
+        logging.debug('             Base Domain: %s' % mgmt_metadata['base_domain'])
+        logging.debug('             AWS Zone:    %s' % mgmt_metadata['aws_region'])
+
+    # Get connected to management cluster
+    if args.mgmt_kubeconfig:
+        mgmt_kubeconfig_path = args.mgmt_kubeconfig
+    elif args.mgmt_cluster:
+        logging.debug('Downloading kubeconfig for Management Cluster %s:' % args.mgmt_cluster)
+        mgmt_kubeconfig_path = _download_kubeconfig(ocm_cmnd,mgmt_metadata['cluster_id'],my_path)
+        if not os.path.exists(mgmt_kubeconfig_path):
+            logging.error('Management Cluster kubeconfig not found %s' % mgmt_kubeconfig_path)
+            exit(1)
+        else:
+            os.environ["KUBECONFIG"] = mgmt_kubeconfig_path
+            # Check if hosted cluster CRD is installed
+            logging.info('Checking if hosted cluster CRD is installed on Management Cluster...')
+            oc_cmd = ["oc", "get", "crd", "hostedclusters.hypershift.openshift.io"]
+            logging.debug(oc_cmd)
+            oc_process = subprocess.Popen(oc_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            oc_stdout,oc_stderr = oc_process.communicate()
+            if oc_process.returncode != 0:
+                logging.error('%s Hosted clusters CRD is not installed on %s' % mgmt_metadata['cluster_name'])
```

Comment:
Script is prepared to be executed from shell or from airflow.
It can use binaries (on airflow, I reuse the binaries already downloaded) or download and compile them.

Scope of the script is just to install the hosted clusters, management cluster needs to be already present. it can connect to it using a kubeconfig file, or passing the name if it is present on ocm and it will download it